client/state/google-apps-users/selectors.js

```diff
@@ -0,0 +1,25 @@
+import createSelector from 'lib/create-selector';
+
+function getGoogleAppsUsersState( state ) {
+	return state.googleAppsUsers;
+}
+
+function createGoogleAppsUsersSelector( fn ) {
+	return createSelector( fn, ( state ) => [ getGoogleAppsUsersState( state ) ] );
+}
+
+export const getByDomain = createGoogleAppsUsersSelector(
+	( state, domainName ) => getGoogleAppsUsersState( state ).items.filter( item => item.domain === domainName )
+);
+
+export const getBySite = createGoogleAppsUsersSelector(
+	( state, siteId ) => getGoogleAppsUsersState( state ).items.filter( item => item.site_id === siteId )
+);
+
+export function getAll( state ) {
```

Comment:
As far as I can tell, most of the existing selectors don't have a `getAll` (actually, none have literally `getAll`) or same in functionality methods. So I don't think adding it is necessary - especially that it's so simple that it can be easily added when needed :)